build.sbt

```diff
@@ -861,6 +869,97 @@ lazy val zioServer: ProjectMatrix = (projectMatrix in file("server/zio-http4s-se
+  .settings(
+    name := "tapir-aws-lambda",
+    libraryDependencies ++= loggerDependencies,
+    libraryDependencies ++= Seq(
+      "com.softwaremill.sttp.client3" %% "http4s-ce2-backend" % Versions.sttp,
+      "org.http4s" %% "http4s-blaze-client" % Versions.http4s
+    )
+  )
+  .jvmPlatform(scalaVersions = allScalaVersions)
+  .dependsOn(core, cats, circeJson, awsSam)
+
+lazy val sam = Process("sam local start-api --warm-containers EAGER").run()
+
+lazy val awsLambdaTests: ProjectMatrix = (projectMatrix in file("serverless/aws/lambda-tests"))
+  .settings(commonJvmSettings)
+  .settings(
+    name := "tapir-aws-lambda-tests",
+    libraryDependencies += "com.amazonaws" % "aws-lambda-java-runtime-interface-client" % Versions.lambdaInterface,
+    assembly / assemblyJarName := "tapir-aws-lambda-tests.jar",
+    assembly / test := {}, // no tests before building jar
+    assembly / assemblyMergeStrategy := {
+      case PathList("META-INF", "io.netty.versions.properties")                    => MergeStrategy.first
+      case _ @("scala/annotation/nowarn.class" | "scala/annotation/nowarn$.class") => MergeStrategy.first
+      case x                                                                       => (assembly / assemblyMergeStrategy).value(x)
+    },
+    Test / test := (Test / test)
+      .dependsOn((Compile / runMain).toTask(" sttp.tapir.serverless.aws.lambda.tests.LambdaSamTemplate"))
+      .dependsOn(assembly)
+      .value,
+    Test / testOptions += Tests.Setup(() => {
+      val log = sLog.value
+      val samReady = PollingUtils.poll(30.seconds, 1.second) {
+        sam.isAlive() && PollingUtils.urlConnectionAvailable(new URL(s"http://127.0.0.1:3000/health"))
+      }
+      if (!samReady) {
+        sam.destroy()
+        log.err("Failed to start sam local-api")
+      }
+    }),
+    Test / testOptions += Tests.Cleanup(() => sam.destroy()),
+    Test / parallelExecution := false
+  )
+  .jvmPlatform(scalaVersions = allScalaVersions)
+  .dependsOn(core, cats, circeJson, awsLambda, awsSam, sttpStubServer, tests, serverTests)
+
+lazy val awsSam: ProjectMatrix = (projectMatrix in file("serverless/aws/sam"))
+  .settings(commonJvmSettings)
+  .settings(
+    name := "tapir-aws-sam",
+    libraryDependencies ++= Seq(
+      "io.circe" %% "circe-yaml" % Versions.circeYaml,
+      "io.circe" %% "circe-generic" % Versions.circe
+    )
+  )
+  .jvmPlatform(scalaVersions = allScalaVersions)
+  .dependsOn(core, tests % Test)
+
+lazy val runSamExample = taskKey[Unit]("runs aws lambda example on sam local")
+
+lazy val awsExamples: ProjectMatrix = (projectMatrix in file("serverless/aws/examples"))
+  .settings(commonJvmSettings)
+  .settings(
+    libraryDependencies += "com.amazonaws" % "aws-lambda-java-runtime-interface-client" % Versions.lambdaInterface
+  )
+  .settings(
+    name := "tapir-aws-examples",
+    assembly / assemblyJarName := "tapir-aws-examples.jar",
+    runSamExample := {
+      val log = sLog.value
```

Comment:
I'm not sure if anybody is actually going to run this example from the sbt sources. Maybe adding this to documentation or a separate, simple runnable project would make more sense. What do you think? And it would have to use the "real" AWS.

Reply:
I think i remove this and describe how to run it in the docs, also for terraform.
That runnable example using real AWS would probably need tasks to build docker image and deploy using sam/terraform. If so then maybe it will be nicer to just put that tasks into the docs, since users would probably just like to copy/paste them to their projects.

Reply:
Yes, seems like a better way :)